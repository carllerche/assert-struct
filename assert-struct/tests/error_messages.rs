// Test error messages with exact format checking
// Each test case is in its own module file for stable line numbers

#[macro_use]
mod util;

error_message_test!(
    "error_messages/simple_field_mismatch.rs",
    simple_field_mismatch
);

error_message_test!(
    "error_messages/nested_comparison.rs",
    nested_comparison
);

error_message_test!(
    "error_messages/equality_pattern.rs",
    equality_pattern
);

error_message_test!(
    "error_messages/range_pattern.rs",
    range_pattern
);

error_message_test!(
    "error_messages/enum_variant.rs",
    enum_variant
);

error_message_test!(
    "error_messages/comparison_pattern.rs",
    comparison_pattern
);

error_message_test!(
    #[cfg(feature = "regex")]
    "error_messages/regex_pattern.rs",
    regex_pattern
);

error_message_test!(
    "error_messages/slice_pattern.rs",
    slice_pattern
);

error_message_test!(
    "error_messages/option_with_comparison.rs",
    option_with_comparison
);
