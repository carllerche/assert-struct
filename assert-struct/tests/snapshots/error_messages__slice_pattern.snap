---
source: assert-struct/tests/error_messages.rs
expression: message
---
assert_struct! failed:

value mismatch:
  --> `data.values.[1]` (assert-struct/tests/error_messages/slice_pattern.rs:13)
  actual: 2
  expected: 5
